Bug: Tool examples broken after PR #380 - need MelleaTool wrapper

## Bug: Tool examples broken after PR #380 - need MelleaTool wrapper

### Description
PR #380 introduced a breaking change requiring `MelleaTool` instances instead of raw callables for `ModelOption.TOOLS`. Several examples and documentation were not updated and now fail.

### Failing Tests
- `docs/examples/tools/interpreter_example.py` ✗ (test fails)

### Affected Documentation (not tested but incorrect)
- `docs/examples/tools/README.md` (lines 45, 56, 83, 103)
- `docs/tutorial.md` (line 1359)

### Error
```
AssertionError: If ModelOption.TOOLS is a list, it must be a list of Tool; found <class 'function'>
```

### Root Cause
The tool system now requires wrapping callables with `MelleaTool.from_callable()`:

**Before (broken):**
```python
from mellea.stdlib.tools import local_code_interpreter

m.instruct(
    description="...",
    model_options={ModelOption.TOOLS: [local_code_interpreter]},
)
```

**After (fixed):**
```python
from mellea.stdlib.tools import local_code_interpreter
from mellea.backends.tools import MelleaTool

m.instruct(
    description="...",
    model_options={ModelOption.TOOLS: [MelleaTool.from_callable(local_code_interpreter)]},
)
```

### Files to Update

**Examples (tested):**
- `docs/examples/tools/interpreter_example.py` (lines 21, 33, 63)

**Documentation (not tested but needs update):**
- `docs/examples/tools/README.md` (lines 45, 56, 83, 103)
- `docs/tutorial.md` (line 1359 - `web_search` function)

**Note:** All tests in `test/backends/` already use `MelleaTool.from_callable()` correctly.

### Additional Context
- PR #380: feat: new MelleaTool class and adoption across mellea
- Commit: ffb8b6c797b20ec387a2c2257e88900b4e498f0f
- See `test/backends/test_mellea_tool.py` for correct usage examples

--- a/docs/examples/tools/README.md
+++ b/docs/examples/tools/README.md
@@ -61,23 +61,26 @@ print(result)  # Output: 2
 ```python
 from mellea import start_session
 from mellea.backends import ModelOption
+from mellea.backends.tools import MelleaTool
 from mellea.stdlib.tools import code_interpreter
 
 m = start_session()
 result = m.instruct(
     "Make a plot of y=x^2",
-    model_options={ModelOption.TOOLS: [code_interpreter]}
+    model_options={ModelOption.TOOLS: [MelleaTool.from_callable(code_interpreter)]}
 )
 ```
 
 ### Forcing Tool Use
 ```python
+from mellea.backends.tools import MelleaTool
 from mellea.stdlib.requirements import uses_tool
+from mellea.stdlib.tools import code_interpreter
 
 result = m.instruct(
     "Use the code interpreter to make a plot of y=x^2",
     requirements=[uses_tool(code_interpreter)],
-    model_options={ModelOption.TOOLS: [code_interpreter]},
+    model_options={ModelOption.TOOLS: [MelleaTool.from_callable(code_interpreter)]},
     tool_calls=True
 )
 
@@ -91,7 +94,9 @@ exec_result = result.tool_calls["code_interpreter"].call_func()
 
 ### Validating Tool Arguments
 ```python
-from mellea.stdlib.requirements import tool_arg_validator
+from mellea.backends.tools import MelleaTool
+from mellea.stdlib.requirements import tool_arg_validator, uses_tool
+from mellea.stdlib.tools import code_interpreter
 
 result = m.instruct(
     "Use the code interpreter to make a plot of y=x^2",
@@ -104,7 +109,7 @@ result = m.instruct(
             validation_fn=lambda code: "/tmp/output.png" in code
         )
     ],
-    model_options={ModelOption.TOOLS: [code_interpreter]},
+    model_options={ModelOption.TOOLS: [MelleaTool.from_callable(code_interpreter)]},
     tool_calls=True
 )
 ```
@@ -119,12 +124,14 @@ result = m.instruct(
 
 Create custom tools by defining functions:
 ```python
+from mellea.backends.tools import MelleaTool
+
 def my_tool(arg1: str, arg2: int) -> str:
     """Tool description for the LLM."""
     return f"Processed {arg1} with {arg2}"
 
 # Use in model_options
-model_options={ModelOption.TOOLS: [my_tool]}
+model_options={ModelOption.TOOLS: [MelleaTool.from_callable(my_tool)]}
 ```
 
 ## Tool Requirements

